oci/spec_opts.go

```diff
@@ -1324,6 +1324,19 @@ func WithLinuxDevices(devices []specs.LinuxDevice) SpecOpts {
 	}
 }
 
+func WithLinuxDeviceSymlink(path, permissions string) SpecOpts {
+	resolvedPath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return func(_ context.Context, _ Client, _ *containers.Container, _ *Spec) error {
+			return err
+		}
+	}
+	path = resolvedPath
+
+	return WithLinuxDevice(path, permissions)
```

Comment:
Perhaps a bit of a "nit" (and personal preference); I prefer `ExportedFunc -> calls -> Unexported` (where possible).

It makes it easier to find if exported functions are actually used "externally" (so in this case, removing `WithLinuxDeviceSymlink` would show that `WithLinuxDevice` is no longer used.

combined with my previous comment; perhaps my suggestion would work? (the function is not "too" complicated yet, so adding the extra boolean arg for a non-exported function seems fine).

Reply:
Yea I agree; I've taken your suggestion in recent revision here